### PR TITLE
chore: Split Text widget

### LINKED ISSUE #29446: Split Text widget
It is necessary to divide the current text widget into two
1. Header — everything is the same as it is now, but the default font size is **heading**
2. Paragraph —  everything is the same as it is now. We just change the names of the widget.

Also, as part of the task, it is necessary to remove the ability to set the **text color**

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #29446 

## Summary by CodeRabbit

- **New Features**
  - Introduced `WDSHeadingWidget` for enhanced heading elements.
  - Launched `WDSParagraphWidget` for improved paragraph formatting.

- **Refactor**
  - Transitioned from `WDSTextWidget` to more specific text-related widgets: `WDSHeadingWidget` and `WDSParagraphWidget`.
  
- **Style**
  - Updated widget styles to reflect new static color assignments and semantic changes.
  
- **Documentation**
  - Updated widget names in configuration files to align with new naming conventions.
  
- **Chores**
  - Removed unused color configuration code from `styleConfig.ts`.
  
- **Tests**
  - Adjusted tests to accommodate new widget structure and naming.

### CODE REVIEW

File: app/client/src/widgets/wds/constants.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ export const WDS_V2_WIDGET_MAP = {
   CHECKBOX_WIDGET: "WDS_CHECKBOX_WIDGET",
   ICON_BUTTON_WIDGET: "WDS_ICON_BUTTON_WIDGET",
   TEXT_WIDGET: "WDS_TEXT_WIDGET",
```
Comment (KelvinOm) [inline app/client/src/widgets/wds/constants.ts:6]: Can we align the widget names here?

**Review by KelvinOm — COMMENTED** [review]
Something is wrong with Heading widget. It is especially worth checking when the provider has a large width.

Снимок экрана 2023-12-21 в 15 12 25

File: app/client/src/widgets/wds/constants.ts (hunk 1 above)
Comment (jsartisan) [reply app/client/src/widgets/wds/constants.ts:6]: Sorry I didn't get you. What do you mean by aligning the names? 🤔

Comment (KelvinOm) [reply app/client/src/widgets/wds/constants.ts:6]: We don't have a text widget anymore. Just a paragraph and a heading.

[Here](https://github.com/appsmithorg/appsmith/pull/29789/files/11ea7277727b6902334c0a3200884da08344ce4f#diff-75c41cc16ba8a3196eda844a9792cfc26087fb3cff448037c01ee3fed2272a45R13) also should be `WDS_PARAGRAPH_WIDGET`, right?

**Pull request merged**